{# Cutoff input page #}

{% extends 'base_templates/base.html' %}

{% block title %}| Cutoff prediction{% endblock %}
{% block nav-cutoff %}active{% endblock %}

{% block container %}
    <head>
        {% load static %}
        <script src="{% static 'js/cutoff.js' %}"></script>
    </head>
    {% include 'base_templates/base_tabs.html' %}
    {% load crispy_forms_filters %}
        <div class="container">
        <form action="{% url 'cutoff_results' %}" method="post" id="id_form"
              enctype="multipart/form-data" target="_blank">
            {% csrf_token %}
            <div id="div_home">
                <h2 class="bg-light border text-center">Cutoff calculation</h2>
                <div class="row">
                    <div class="col">
                        <div class="card h-100">
                            <h5 class="card-header">Reference data input</h5>
                                <div class="card-body">
                                    {{ form.input_file|as_crispy_field }}
                                    {{ form.sim_file|as_crispy_field }}
                                </div>
                        </div>
                    </div>

                    <div class="col">
                        <div class="card h-100">
                            <h5 class="card-header">Type of cut-off</h5>
                            <div class="card-body">
                                {{ form.cutoff_type|as_crispy_field }}
                            </div>
                        </div>
                    </div>

                    <div class="col">
                        <div class="card h-100">
                            <h5 class="card-header">Rank</h5>
                            <div class="card-body">
                                {{ form.rank|as_crispy_field }}
                                {{ form.higher_rank|as_crispy_field }}
                            </div>
                        </div>
                    </div>


                </div>
                <div class="d-grid gap-3 col-2 mx-auto mt-30">
                    <button type="submit" class="btn btn-primary">Submit</button>
                </div>
            </div>

            <div id="div_settings" class="hidden">
                <h2 class="bg-light border text-center">Settings</h2>
                <div class="row align-items-end">
                    <div class="col">
                        {{ form.max_seq_number|as_crispy_field }}
                    </div>
                    <div class="col">
                        {{ form.min_group_number|as_crispy_field }}
                    </div>
                </div>
                <div class="row align-items-end">
                    <div class="col">
                        {{ form.min_alignment_length|as_crispy_field }}
                    </div>
                    <div class="col">
                        {{ form.min_seq_number|as_crispy_field }}
                    </div>
                </div>
                <div class="row">
                    <div class="col">
                        <label class="form-label">Thresholds and step size of prediction:</label>
                        <div class="input-group d-flex flex-nowrap">
                                {{ form.starting_threshold|as_crispy_field }}
                                {{ form.end_threshold|as_crispy_field }}
                                {{ form.step|as_crispy_field }}
                        </div>

                    </div>
                    <div class="col">
                        <div class="card">
                            <ul class="list-group list-group-flush">
                                <li class="list-group-item">
                                    <div class="form-check form-switch">
                                        {{ form.remove_comp|as_crispy_field }}
                                    </div>
                                </li>
                                <li class="list-group-item">
                                    {{ form.cutoff_remove|as_crispy_field }}
                                </li>
                            </ul>
                        </div>
                    </div>
                </div>
            </div>
        </form>
        <div id="div_explanation">
            <h3 class="bg-light border text-center mt-50">How to use this page</h3>
            <p class="mt-30">
                Welcome to DNA barcoder and thank you for choosing this application for your metabarcoding classification!<br>
                Interested in how this application is different from the others?
                Go to the <a href="{% url 'about' %}">about</a> page!
            </p>

            <h4>Reference data input</h4>
            <p>
                The reference dataset can be given as a file with FASTA format.<br>
                Make sure the description lines of the FASTA file have the following format:<br>
                <code>>ID_NAME k__kingdom;p__phylum;c__class;o__order;f__family;s__species</code><br>
                Example:<br>
                <code>>MH854570 k__Fungi;p__Ascomycota;c__Sordariomycetes;o__Hypocreales;f__Nectriaceae;g__Fusarium;s__Fusarium_equiseti</code>
            </p>

            <h4>Type of cutoff</h4>
            <p>
                The similarity cutoff is a percentage at which an unidentified sequence and a reference sequence have to minimally coincide.
                This can be given as a global value or local values. A global similarity cutoff is representative for the whole dataset.
                Local similarity cutoffs are given per taxon. Local similarity cutoff wills generally give
                more accurate results (see the <a href="{% url 'about' %}">about</a> page).<br>
            </p>

            <h4>Rank</h4>
            <p>
                The identification rank is the taxonomic level at which will be classified.<br>
                When calculating local cutoff values, a higher rank as to be selected.
                A cutoff value will be calculated for each clade of that taxonomic level.
            </p>

            <h4>Settings</h4>
            <p>
                <strong>Note for the <i>Minimum sequence alignment length</i> setting:</strong><br>
                The default for this setting is 400. It is however recommended
                lowering this when working with smaller barcodes.<br>
                When using ITS sequences it is recommended to use 50.<br>
                <br>
                <strong>Note for the <i>Minimum number of groups</i> and <i>Minimum number of sequences</i> settings:</strong><br>
                The <i>minimum number of groups</i> setting refers to the number of different clades of the <i>identification rank</i> within a <i>higher rank</i> clade.<br>
                The <i>minimum number of sequences</i> setting refers to the number of sequences within a <i>higher rank</i> clade.<br>
                When working with smaller datasets it is recommended to lower these settings.
            </p>
        </div>
        </div>
{% endblock %}
